File: apple/packages/EnteComponents/Sources/EnteComponents/Sheet.swift
import SwiftUI

public struct SheetContent<Content: View>: View {
    @Environment(\.entePalette) private var palette
    private let title: String
    private let message: String?
    private let back: (() -> Void)?
    private let dismiss: (() -> Void)?
    private let backLabel: String
    private let dismissLabel: String
    private let content: Content

    public init(
        title: String,
        message: String? = nil,
        back: (() -> Void)? = nil,
        dismiss: (() -> Void)? = nil,
        backLabel: String = "Back",
        dismissLabel: String = "Close",
        @ViewBuilder content: () -> Content,
    ) {
        self.title = title
        self.message = message
        self.back = back
        self.dismiss = dismiss
        self.backLabel = backLabel
        self.dismissLabel = dismissLabel
        self.content = content()
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: EnteSpacing.lg) {
            SheetHeader(title) {
                if let back {
                    IconAction(kind: .circular, size: 40, action: back) {
                        Glyph.back.image.resizable().scaledToFit()
                            .frame(width: 24, height: 24)
                            .flipsForRightToLeftLayoutDirection(true)
                    }
                    .accessibilityLabel(backLabel)
                }
            } trailing: {
                if let dismiss {
                    IconAction(kind: .circular, size: 40, action: dismiss) {
                        Glyph.close.image.resizable().scaledToFit()
                            .frame(width: 18, height: 18)
                    }
                    .accessibilityLabel(dismissLabel)
                }
            }
            if let message {
                Text(message)
                    .font(EnteTypography.body)
                    .foregroundStyle(palette.mutedText)
            }
            content
        }
        .padding(EnteSpacing.xl)
        .background(palette.background)
    }
}

File: android/packages/components/src/main/kotlin/io/ente/components/Sheet.kt
package io.ente.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp

@Composable
public fun SheetContent(
    title: String,
    modifier: Modifier = Modifier,
    message: String? = null,
    back: (() -> Unit)? = null,
    dismiss: (() -> Unit)? = null,
    backLabel: String = "Back",
    dismissLabel: String = "Close",
    content: @Composable () -> Unit,
) {
    val palette = LocalEntePalette.current
    Column(modifier.background(palette.background).padding(EnteSpacing.xl)) {
        SheetHeader(
            title,
            leading =
                back?.let { onBack ->
                    {
                        IconAction(onBack, kind = IconActionKind.Circular) {
                            Icon(
                                painterResource(R.drawable.component_back),
                                backLabel,
                                Modifier.size(24.dp),
                            )
                        }
                    }
                },
            trailing =
                dismiss?.let { onDismiss ->
                    {
                        IconAction(onDismiss, kind = IconActionKind.Circular) {
                            Icon(
                                painterResource(R.drawable.component_close),
                                dismissLabel,
                                Modifier.size(18.dp),
                            )
                        }
                    }
                },
        )
        Spacer(Modifier.height(EnteSpacing.lg))
        if (message != null) {
            Text(message, style = EnteTypography.body, color = palette.mutedText)
            Spacer(Modifier.height(EnteSpacing.lg))
        }
        content()
    }
}
